src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets

```diff
@@ -70,6 +70,7 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <Target Name="GenerateBuildDependencyFile"
           DependsOnTargets="_DefaultMicrosoftNETPlatformLibrary"
+          BeforeTargets="_CheckForCompileOutputs"
```

Comment:
My instinct would be to go before `_CleanGetCurrentAndPriorFileWrites` instead, since this isn't really compile-specific. But `_CheckForCompileOutputs` is before that so this is ok. Did you prefer this for a reason?